files/lib/system/endpoint/controller/faq/questions/search/GetSearch.class.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace wcf\system\endpoint\controller\faq\questions\search;
+
+use Laminas\Diactoros\Response\JsonResponse;
+use Override;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use wcf\data\faq\Question;
+use wcf\data\faq\QuestionList;
+use wcf\http\Helper;
+use wcf\system\endpoint\GetRequest;
+use wcf\system\endpoint\IController;
+use wcf\system\exception\UserInputException;
+use wcf\system\WCF;
+
+#[GetRequest('/faq/questions/search')]
+final class GetSearch implements IController
+{
+    #[Override]
+    public function __invoke(ServerRequestInterface $request, array $variables): ResponseInterface
+    {
+        $parameters = Helper::mapApiParameters($request, GetSearchParameters::class);
+        if (\mb_strlen($parameters->query) < 3) {
+            throw new UserInputException('query', 'tooShort');
+        }
+
+        $questionIDs = $this->getQuestionsIDs($parameters->query);
+
+        return new JsonResponse([
+            'template' => WCF::getTPL()->fetch('faqQuestionSearchResult', 'wcf', [
+                'questions' => $this->getQuestions($questionIDs),
+            ], true),
+        ]);
+    }
+
+    /**
+     * @return list<int>
+     */
+    private function getQuestionsIDs(string $query): array
+    {
+        $sql = "
+                SELECT          faq_questions.questionID
+                FROM            wcf1_faq_questions faq_questions
+                LEFT JOIN       wcf1_language_item language_item
+                            ON	language_item.languageItem = faq_questions.question
+                WHERE           faq_questions.question LIKE ?
+                            OR	(
+                                    language_item.languageItemValue LIKE ?
+                                AND	language_item.languageID = ?
+                                )
+                ORDER BY		faq_questions.question
+            ";
+        $statement = WCF::getDB()->prepare($sql, 5);
+        $statement->execute([
+            '%' . $query . '%',
+            '%' . $query . '%',
```

Comment:
`$query` should be sanitized using `WCF::getDB()->escapeLikeValue()` which prevents the special characters `\`, `%` and `_` to be interpreted.